sparkmagic/sparkmagic/tests/test_kernel_magics.py

```diff
@@ -402,6 +402,33 @@ def test_configure():
     magic.info.assert_called_once_with("")
 
 
+
+@with_setup(_setup, _teardown)
+@patch('hdijupyterutils.configuration._load', return_value = {'required_session_configs': {'conf': {'spark.yarn.tags': "created-by=vaatu-raava"}}})
+def test_configure_with_required_settings(hdijupyterutils_load):
+    # Mock info method
+    magic.info = MagicMock()
+    # Session not started
+    conf.override_all({})
+    magic.configure('', '{"extra": "yes"}')
+    assert_equals(conf.session_configs(), {"extra": "yes", 'conf': {'spark.yarn.tags': "created-by=vaatu-raava"}})
+    _assert_magic_successful_event_emitted_once('configure')
+    magic.info.assert_called_once_with("")
+
+
+@with_setup(_setup, _teardown)
+@patch('hdijupyterutils.configuration._load', return_value = {'required_session_configs': {}})
```

Comment:
What about the case where both `session_configs` and `required_session_configs` define `'{"extra": ...}'`?

Reply:
i can add a test here, but just to make sure you also feel the same way about how this merge works, I'd expect the required session configs to override extra session configs in a nested fashion

Reply:
Agreed - thats my expectation as well

Reply:
added please take a look @jpugliesi